test/unit/kubernetes-deploy/kubernetes_resource_test.rb

```diff
@@ -288,6 +288,17 @@ def test_disappeared_is_false_if_resource_has_been_deployed_and_we_get_a_server_
     refute_predicate(dummy, :disappeared?)
   end
 
+  def test_custom_resource_kind
+    definition = { "kind" => "foobar", "metadata" => { "name" => "test" } }
+    KubernetesDeploy::KubernetesResource.build(
+      namespace: 'test',
+      context: 'test',
+      definition: definition,
+      logger: ::Logger.new($stderr),
```

Comment:
Not that it makes any difference here, but FYI there's a `logger` accessor all tests can use.